docs/version-format.md

```diff
@@ -0,0 +1,39 @@
+Version Formats
+===============
+
+In some circumstances fwupd has to convert from a unsigned integer version
+number into something that has either been used in documentation or has been
+defined in some specification.
+A good example here is the UEFI ESRT table, which specifies a `uint32_t` for
+the version but does not specify how this should be formatted for the user.
+
+As is typical in underspecified specifications, vendors have converted the
+integer in different ways. For instance, Dell uses version strings like 1.2.3
+and Microsoft use versions like 1.2.3.4.
+
+The fwudp daemon can match specific devices and apply the correct version style
+using quirk files. The version format can also be specified in the firmware
+`metainfo.xml` file so that the new version is correctly shown, and so that it
+matches on the LVFS website.
+
+The current version formats supported by fwupd and the LVFS are:
+
+ * `plain`:     Use plain integer version numbers with no dots, e.g. `AABBCCDD`
+ * `quad`:      Use Dell-style `AA.BB.CC.DD` version numbers
+ * `triplet`:   Use Microsoft-style `AA.BB.CCDD` version numbers
+ * `pair`:      Use two `AABB.CCDD` version numbers
+ * `bcd`:       Use binary coded decimal notation
+ * `intel-me`:  Use Intel ME-style notation (`aaa+11.bbbbb.CC.DDDD`)
+ * `intel-me2`: Use Microsoft-style `A.B.CC.DDDD` notation
+
+These can be specified in quirk files like this:
+
+    # Vendor Modelname
+    [DeviceInstanceId=5b92717b-2cad-4a96-a13b-9d65781df8bf]
```

Comment:
It might be clearer to also show that GUID can be used interchangably for DeviceInstanceID